src/zenml/config/global_config.py

```diff
@@ -106,20 +430,254 @@ def _read_config(self) -> Dict[str, Any]:
 
         return config_values
 
-    def _write_config(self) -> None:
-        """Writes the global configuration options to disk."""
+    def _write_config(self, config_path: Optional[str] = None) -> None:
+        """Writes the global configuration options to disk.
+
+        Args:
+            config_path: custom config file path. When not specified, the default
+                global configuration path is used.
+        """
+        config_file = self._config_file(config_path)
         yaml_dict = json.loads(self.json())
-        yaml_utils.write_yaml(self.config_file(), yaml_dict)
+        logger.debug(f"Writing config to {config_file}")
+
+        if not fileio.file_exists(config_file):
+            fileio.create_dir_recursive_if_not_exists(
+                config_path or self.config_directory
+            )
+
+        yaml_utils.write_yaml(config_file, yaml_dict)
 
     @staticmethod
-    def config_directory() -> str:
-        """Path to the global configuration directory."""
+    def default_config_directory() -> str:
+        """Path to the default global configuration directory."""
         return get_global_config_directory()
 
-    @staticmethod
-    def config_file() -> str:
-        """Path to the file where global configuration options are stored."""
-        return os.path.join(GlobalConfig.config_directory(), "config.yaml")
+    def _config_file(self, config_path: Optional[str] = None) -> str:
+        """Path to the file where global configuration options are stored.
+
+        Args:
+            config_path: custom config file path. When not specified, the default
+                global configuration path is used.
+        """
+        return os.path.join(config_path or self._config_path, "config.yaml")
+
+    def copy_config_with_active_profile(
```

Comment:
I like the idea of this function, but I feel like it could be made simpler by just always storing the contents of the profile into a new YAML_STORAGE based one, instead of copying along relative paths from who knows where and adjusting the urls to those.